cli/src/commands/init/index.ts

```diff
@@ -1,17 +1,292 @@
+      message: 'Do you want to install the TypeScript/JavaScript SDK?',
+      initial: true
+    });
+    if (!confirm) return;
+
+    await this.installPackage('@xata.io/client');
+
+    await this.configureCodegen();
+  }
+
+  async configureCodegen() {
+    const { confirm } = await prompts({
+      type: 'confirm',
+      name: 'confirm',
+      message: 'Do you want to use the TypeScript/JavaScript code generator?',
+      initial: true
+    });
+    if (!confirm) return;
+
+    this.projectConfig = this.projectConfig || {};
+    this.projectConfig.codegen = {};
+
+    const { output } = await prompts({
+      type: 'text',
+      name: 'output',
+      message: 'Choose where the output file for the code generator',
+      initial: 'src/xata.ts'
+    });
+    if (!output) return this.error('You must provide an output file');
+
+    this.projectConfig.codegen.output = output;
+
+    if (!output.endsWith('.ts')) {
+      const { declarations } = await prompts({
+        type: 'confirm',
+        name: 'declarations',
+        message: 'Do you want to generate the TypeScript declarations?',
+        initial: (prev) => !prev.endsWith('.ts')
+      });
+
+      if (declarations) {
+        this.projectConfig.codegen.declarations = true;
+      }
+    }
+  }
+
+  async getPackageManager() {
+    const packageManager = (await this.access('yarn.lock')) ? 'yarn' : 'npm';
+    if (!which.sync(packageManager, { nothrow: true })) {
+      this.error(
+        `Looks like ${packageManager} is not installed or is not in the PATH. This made impossible to install the code generator`
+      );
+    }
+    return packageManager;
+  }
+
+  async access(path: string) {
+    try {
+      await access(path);
+      return true;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  async installPackage(pkg: string) {
+    const command = await this.getPackageManager();
+    const subcommand = command === 'yarn' ? 'add' : 'install';
+    await this.runCommand(command, [subcommand, pkg]);
+  }
+
+  runCommand(command: string, args: string[]) {
+    this.log(`Running ${command} ${args.join(' ')}`);
+    return new Promise((resolve, reject) => {
+      spawn(command, args, { stdio: 'inherit' }).on('exit', (code) => {
+        if (code && code > 0) return reject(new Error('Command failed'));
+        resolve(undefined);
+      });
+    });
+  }
+
+  async writeConfig() {
+    // Reuse location when using --force
+    if (!this.projectConfigLocation) {
+      const { location } = await prompts({
+        type: 'select',
+        name: 'location',
+        message: 'Select where to store your configuration',
+        choices: this.searchPlaces.map((place) => ({ title: place, value: place }))
+      });
+      if (!location) return this.error('You must select a location for the configuration file');
+
+      this.projectConfigLocation = path.join(process.cwd(), location);
+    }
+    await this.updateConfig();
+  }
+
+  async writeEnvFile() {
```

Comment:
We could offer a command to generate/update API Key in the ``.env`` file but that doesn't require you to run ``init``